### PR TITLE
Story1

### PR DESCRIPTION
**Pull request by jennymegan**
"add item" page completed.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -34,8 +34,24 @@ function populateTable(array $vinylDetails): string {
             isset($vinyl['year'])
             ) {
 
+<<<<<<< HEAD
```
Comment (mporam) [inline functions.php]: i dont think this is supposed to be here?

File: newEntry.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <div class="logo">
+        <img src="logo1.png" alt="Jenny Logo">
+    </div>
+    <h1>MY VINYL COLLECTION</h1>
+    <div></div>
+</div>
+<div class="container">
+    <form action="newEntryHandling.php" enctype='multipart/form-data' method="POST">
+        <ul class=""form_wrapper">
+        <li class="form_row"><label>Artist First Name*: </label><input type="text" name="artist_firstname" required></li>
```
Comment (mporam) [inline newEntry.php]: these lines are very long, drop the labels onto their own line and the inputs

File: newEntry.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <li class="form_row"><label>Album*: </label><input type="text" name="album" required></li>
+        <li class="form_row"><label>Release Year (YYYY)*: </label><input type="text" name="year" maxlength="4" required></li>
+        <li class="form_row"><label>Album Art: </label><label class="file_upload"><input class="file_upload" type="file" name="cover_art"></label></li>
+        <li class="form_row"><input class="add_vinyl_button" type = "submit" name="upload" value="Add My Vinyl!"></li>
+        </ul>
+    </form>
+    <div class="error">
+        <h4><?php
+        if (isset($_SESSION['Error'])) {
+        echo $_SESSION['Error'];
```
Comment (mporam) [inline newEntry.php]: things inside an if statement should be indented

File: newEntry.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        <li class="form_row"><label>Artist Last Name: </label><input type="text" name="artist_lastname"></li>
+        <li class="form_row"><label>Album*: </label><input type="text" name="album" required></li>
+        <li class="form_row"><label>Release Year (YYYY)*: </label><input type="text" name="year" maxlength="4" required></li>
+        <li class="form_row"><label>Album Art: </label><label class="file_upload"><input class="file_upload" type="file" name="cover_art"></label></li>
+        <li class="form_row"><input class="add_vinyl_button" type = "submit" name="upload" value="Add My Vinyl!"></li>
+        </ul>
+    </form>
+    <div class="error">
+        <h4><?php
+        if (isset($_SESSION['Error'])) {
```
Comment (mporam) [inline newEntry.php]: sessions are not a great place to store error messages, have a look at joshs code to see how to do it with a GET query

File: newEntryHandling.php [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+session_start();
+include ('functions.php');
+
+$db = getDatabase('collection_app');
+
+if(
+    !empty($_POST['artist_firstname']) &&
+    !empty($_POST['album']) &&
+    !empty($_POST['year'])
+    ){
```
Comment (mporam) [inline newEntryHandling.php]: this should be at the same level as your if statement

File: newEntryHandling.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<?php
+session_start();
+include ('functions.php');
+
+$db = getDatabase('collection_app');
+
+if(
```
Comment (mporam) [inline newEntryHandling.php]: missing space after the `if`

File: newEntryHandling.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    !empty($_POST['artist_firstname']) &&
+    !empty($_POST['album']) &&
+    !empty($_POST['year'])
+    ){
+        if(!empty($_FILES['cover_art']['name'])){
```
Comment (mporam) [inline newEntryHandling.php]: PSR!

File: newEntryHandling.php [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+            $name = $_FILES['cover_art']['name'];
+            $target_dir = dirname(__FILE__); //Gets current filepath (one i'm working in)
```
Comment (mporam) [inline newEntryHandling.php]: remove comments

File: newEntryHandling.php [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+            // Check file type is ok
+            $imageFileType = strtolower(pathinfo($target_file,PATHINFO_EXTENSION));
+            $extensions_arr = ["jpg","jpeg","png","gif"];
+
+            if(in_array($imageFileType,$extensions_arr) ){
+                move_uploaded_file($_FILES['cover_art']['tmp_name'], $target_file);
+
+                //Upload data with cover image
+                $query=$db->prepare('INSERT INTO `my_vinyl_collection` (`artist_firstname`, `album`, `year`, `artist_lastname`, `cover_art`) VALUES (:firstname, :album, :year, :lastname, :art); ');
+                $query->execute(['firstname'=>$_POST['artist_firstname'], 'album' => $_POST['album'], 'year' => $_POST['year'],'lastname' =>$_POST['artist_lastname'], 'art' => $name]);
```
Comment (mporam) [inline newEntryHandling.php]: speak to ellie about how to tidy this up

File: newEntryHandling.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+            } else {
+                $_SESSION['Error'] = '<br><br><h2>You attempted to upload the incorrect filetype. Please try again. </h2><br><br>';
+                header('Location: newEntry.php');
+                exit;
+            }
+} else {
+    //upload data with no cover image
+    $query=$db->prepare('INSERT INTO `my_vinyl_collection` (`artist_firstname`, `album`, `year`, `artist_lastname`) VALUES (:firstname, :album, :year, :lastname); ');
+    $query->execute(['firstname'=>$_POST['artist_firstname'], 'album' => $_POST['album'], 'year' => $_POST['year'],'lastname' =>$_POST['artist_lastname']]);
```
Comment (mporam) [inline newEntryHandling.php]: speak to ellie about how to tidy this up too

**Review by mporam — CHANGES_REQUESTED** [review]

File: functions.php [hunk 11]
```diff
@@ -69,4 +51,74 @@ function populateTable(array $vinylDetails): string {
...
+function addNewVinylNoArt(array $vinylArray,PDO $db){
+
+    if (
+        !empty($vinylArray['artist_firstname']) &&
+        !empty($vinylArray['album']) &&
+        !empty($vinylArray['year'])
+    ) {
+        $vinylArray['artist_lastname'] = $vinylArray['artist_lastname'] ?? '';
+        $query = $db->prepare('INSERT INTO `my_vinyl_collection` (`artist_firstname`, `album`, `year`, `artist_lastname`) VALUES (:artist_firstname, :album, :year, :artist_lastname); ');
+        $query->execute($vinylArray);
```
Comment (mporam) [inline functions.php]: what if the query fails?

File: functions.php [hunk 12]
```diff
@@ -69,4 +51,74 @@ function populateTable(array $vinylDetails): string {
...
+            $extensions_arr = ["jpg", "jpeg", "png", "gif"];
+
+            if (in_array($imageFileType, $extensions_arr)) {
+                move_uploaded_file($file['cover_art']['tmp_name'], $target_file);
+
+                //Upload data with cover image
+                $vinylArray['artist_lastname'] = $vinylArray['artist_lastname'] ?? '';
+                $vinylArray['cover_art'] = $name;
+                $query = $db->prepare('INSERT INTO `my_vinyl_collection` (`artist_firstname`, `album`, `year`, `artist_lastname`, `cover_art`) VALUES (:artist_firstname, :album, :year, :artist_lastname, :cover_art); ');
+                $query->execute($vinylArray);
```
Comment (mporam) [inline functions.php]: no checks if the query failed?

File: newEntryHandling.php [hunk 13]
```diff
@@ -1,43 +1,17 @@
...
-
-                //Upload data with cover image
-                $query=$db->prepare('INSERT INTO `my_vinyl_collection` (`artist_firstname`, `album`, `year`, `artist_lastname`, `cover_art`) VALUES (:firstname, :album, :year, :lastname, :art); ');
-                $query->execute(['firstname'=>$_POST['artist_firstname'], 'album' => $_POST['album'], 'year' => $_POST['year'],'lastname' =>$_POST['artist_lastname'], 'art' => $name]);
-            } else {
-                $_SESSION['Error'] = '<br><br><h2>You attempted to upload the incorrect filetype. Please try again. </h2><br><br>';
-                header('Location: newEntry.php');
-                exit;
-            }
+if (empty($_FILES['cover_art']['name'])) {
```
Comment (mporam) [inline newEntryHandling.php:7]: there is a better way of doing this so you reduce the amount of duplicate code and dont need 2 functions, but its a not insignificant re-write so i wont make you do it

**Review by mporam — CHANGES_REQUESTED** [review]

File: newEntryHandling.php (hunk 13 above)
Comment (jennymegan) [reply newEntryHandling.php:7]: that's very kind. but maybe at some point you could explain it to me so next time I do it more betterer.

**Review by mporam — APPROVED** [review]

**Pull request merged**